src/pytorch_lightning/utilities/data.py

```diff
@@ -350,17 +382,36 @@ def _dataloader_init_kwargs_resolve_sampler(
     return {"sampler": sampler, "shuffle": False, "batch_sampler": None}
 
 
+def _inject_arg_to_saved(
+    args: List[Any], kwargs: Dict[str, Any], arg_names: List[str], inject_name: str, inject_object: Any
+) -> Tuple[bool, List[Any], Dict[str, Any]]:
+    """Tries to inject a custom argument to a saved list of args and kwargs.
+
+    Returns a tuple indicating success of the operation and modified saved args and kwargs
+    """
+
+    if inject_name in arg_names:
+        inject_index = arg_names.index(inject_name)
+        args = args[:inject_index] + [inject_object] + args[inject_index + 1 :]
+        return True, args, kwargs
+    elif inject_name in kwargs:
+        kwargs[inject_name] = inject_object
+        return True, args, kwargs
+
+    return False, args, kwargs
+
+
 def _auto_add_worker_init_fn(dataloader: DataLoader, rank: int) -> None:
     if int(os.environ.get("PL_SEED_WORKERS", 0)) and dataloader.worker_init_fn is None:
         dataloader.worker_init_fn = partial(pl_worker_init_function, rank=rank)
 
 
-def _wrap_dataloader_init(init: Callable) -> Callable:
```

Comment:
Do you think it would be useful to keep this function name, calling _wrap_init_method where `dataset`is passed as the explicit arg?

Reply:
Umm... no. I don't see an added value in keeping around multiple specialized functions, when they can be made one generic function while changing one or two arguments